### PR TITLE
Convert point clouds to generate multi-dimensional data

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Comment by aprokop** [comment]
retest this please

File: benchmarks/point_clouds/point_clouds.hpp [hunk 1]
```diff
@@ -29,6 +29,9 @@ enum class PointCloudType
   hollow_sphere
 };
 
+namespace
+{
```
[the remark this answers is not in the document]
Comment (aprokop) [reply benchmarks/point_clouds/point_clouds.hpp:33]: Hmm, I think I wanted to make sure that it does not cause duplicated symbols even if the non-templated functions were inlined.

File: benchmarks/point_clouds/point_clouds.hpp [hunk 2]
```diff
@@ -75,59 +78,23 @@ void hollowBoxCloud(
...
   unsigned int const n = random_points.extent(0);
-  for (unsigned int i = 0; i < n; ++i)
+  constexpr auto DIM = ArborX::GeometryTraits::dimension<Point>::value;
+  for (unsigned int i = 0; i < n;)
   {
-    unsigned int face = i % 6;
-    switch (face)
-    {
-    case 0:
+    for (int face = 0; face < 2 * DIM && i < n; ++face, ++i)
```
[the remark this answers is not in the document]
Comment (aprokop) [reply benchmarks/point_clouds/point_clouds.hpp]: I rewrite it a bit to hopefully make things more clear.

File: benchmarks/point_clouds/point_clouds.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rombur) [reply benchmarks/point_clouds/point_clouds.hpp:33]: Let's not do that it's against the cpp core guideline https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed

[the remark this answers is not in the document]
Comment (aprokop) [reply benchmarks/point_clouds/point_clouds.hpp]: Fixed.

**Review by Rombur — APPROVED** [review]

**Pull request merged**